pointcloud/basic_octree_utils.go

```diff
@@ -165,11 +161,41 @@ func (octree *basicOctree) helperIterate(lowerBound, upperBound, idx int, fn fun
 			idx += numPoints
 		}
 
-	case LeafNodeFilled:
+	case leafNodeFilled:
 		ok = fn(octree.node.point.P, octree.node.point.D)
 
-	case LeafNodeEmpty:
+	case leafNodeEmpty:
 	}
 
 	return ok
 }
+
+// ConvertPointCloudToBasicOctree converts a pointcloud into a octree pointcloud representation.
+// TBD, add test case
+func ConvertPointCloudToBasicOctree(cloud PointCloud) (PointCloud, error) {
+	center := r3.Vector{
+		X: cloud.MetaData().MinX + (cloud.MetaData().MaxX-cloud.MetaData().MinX)/2,
+		Y: cloud.MetaData().MinY + (cloud.MetaData().MaxY-cloud.MetaData().MinY)/2,
+		Z: cloud.MetaData().MinZ + (cloud.MetaData().MaxZ-cloud.MetaData().MinZ)/2,
+	}
+
+	maxSideLength := math.Max((cloud.MetaData().MaxX - cloud.MetaData().MinX),
```

Comment:
Same question here. the maxSideLength is just a function of the `cloud.MetaData()` could we make it a method on `PointCloud` or a helper function that takes a `cloud.MetaData()`?